src/Tools/dotnet-monitor/GenerateApiKeyCommandHandler.cs

```diff
@@ -18,39 +22,84 @@ namespace Microsoft.Diagnostics.Tools.Monitor
+            outputBldr.AppendLine();
+
+            RootOptions opts = new RootOptions()
             {
-                console.Error.WriteLine(
-                    string.Format(
-                        CultureInfo.CurrentCulture, 
-                        Strings.ErrorMessage_ParameterNotAllowed, 
-                        nameof(hashAlgorithm), 
-                        hashAlgorithm));
-                return Task.FromResult(1);
-            }
+                Authentication = new AuthenticationOptions()
+                {
+                    MonitorApiKey = new MonitorApiKeyOptions()
+                    {
+                        Subject = newJwt.Subject,
+                        PublicKey = newJwt.PublicKey,
+                    }
+                }
+            };
 
-            if (keyLength < ApiKeyAuthenticationHandler.ApiKeyByteMinLength || keyLength > ApiKeyAuthenticationHandler.ApiKeyByteMaxLength)
+            outputBldr.AppendFormat(CultureInfo.CurrentCulture, Strings.Message_SettingsDump, output);
+            outputBldr.AppendLine();
+            switch (output)
             {
-                console.Error.WriteLine(
-                    string.Format(
-                        CultureInfo.CurrentCulture,
-                        Strings.ErrorMessage_ParameterNotAllowedByteRange,
-                        nameof(hashAlgorithm),
-                        hashAlgorithm,
-                        ApiKeyAuthenticationHandler.ApiKeyByteMinLength,
-                        ApiKeyAuthenticationHandler.ApiKeyByteMaxLength));
-                return Task.FromResult(1);
+                case OutputFormat.Json:
+#if NET5_0_OR_GREATER
+                    string optsJson = JsonSerializer.Serialize(opts, new JsonSerializerOptions() 
+                    { 
+                        DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+                        WriteIndented = true 
+                    });
+#else
+                    string optsJson = JsonSerializer.Serialize(opts, new JsonSerializerOptions() 
+                    { 
+                        IgnoreNullValues = true,
+                        WriteIndented = true 
+                    });
+#endif
+                    outputBldr.AppendLine(optsJson);
+                    break;
+                case OutputFormat.Text:
+                    outputBldr.AppendLine(FormattableString.Invariant($"Subject: {newJwt.Subject}"));
+                    outputBldr.AppendLine(FormattableString.Invariant($"PublicKey: {newJwt.PublicKey}"));
+                    break;
+                case OutputFormat.Cmd:
+                case OutputFormat.PowerShell:
+                case OutputFormat.Bash:
+                    IDictionary<string, string> optList = opts.ToEnvironmentConfiguration();
+                    foreach (KeyValuePair<string, string> kvp in optList)
```

Comment:
nit: This is a great candidate for using a deconstructor:

```suggestion
                    foreach ((string name, string value) in optList)
```

Reply:
This is my new favorite language sugar.